### ISSUE TITLE
Webpack Config in .eslintrc

### ISSUE THREAD

**Issue by aolsenjazz** [electron-react-boilerplate/electron-react-boilerplate#3086]
## What is this webpack config in .eslintrc doing?
What is the purpose of the following config in .eslintrc -> module.exports -> settings -> 'import/resolver'?
````
...
webpack: {
  config: require.resolve('./.erb/configs/webpack.config.eslint.ts'),
},
...
````

I ask because this extra configuration breaks eslint when using aliases. By removing this config, aliases work as expected. Building, starting, and packaging are all unaffected.

I've scoured google to find out what packages might be using this and have come up dry. Any info would be super appreciated so I know I'm not unknowingly breaking anything. Thank you again for all your work!

**Comment by amilajack** [comment]
`eslint-plugin-import` was breaking in certain eslint execution contexts, where the relative dir wasn't the root dir.

Previously we were doing something like:

```
webpack: {
  config: "./.erb/configs/webpack.config.eslint.ts"
},
```

`require.resolve` ensures we get an absolute path, and are always able to resolve the correct path.

 Closing but lmk if you have any more questions.

**Comment by aolsenjazz** [comment]
I wasn't clear with how I phrased the question sorry. I'm not questioning the `require.resolve` but the existence of the webpack (and config) key.

Phrased differently: which plugin is using this bit of configuration, and for what?

**Comment by amilajack** [comment]
I see, the `webpack` config key is used to resolve webpack dependencies.

The plugin provides the resolution mechanisms (defined by webpack) to eslint. In the context of aliases, the config allows eslint to lint imports by using your webpack config to resolve your imports.

In your case, eslint will use your webpack config to resolve aliases defined by webpack.

**Comment by aolsenjazz** [comment]
Gotcha, thanks for the clarification.

I'm afraid that imports aren't being linted with the current boilerplate. I was able to get import linting to work for all files in src directory (but _not_ webpack config files) by doing the following:

1. install [eslint-import-resolver-typescript](https://www.npmjs.com/package/eslint-import-resolver-typescript)
2. add 'plugin:import/recommended' to eslintrc -> extends
3. remove aforementioned webpack config snippet. this snippet causes eslint to spit out a smorgasbord of other errors unrelated to imports when trying to resolve import paths.

The smorgasbord of errors seems to be typescript failing to parse the webpack config files. I've included a bit of its output below. If any of this is helpful, I'm happy to open a PR or create a demo repo!

````
Error resolving webpackConfig .erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~

````

**Comment by amilajack** [comment]
I see, thanks for catching this. Fixed in https://github.com/electron-react-boilerplate/electron-react-boilerplate/commit/f242b675a70c1366495d20562be176ca9298b739. Lmk if this change works for you.

**Comment by aolsenjazz** [comment]
Just tested. eslint now catches errors with import paths, but the output is still very unhelpful:

alexanderolsen@Alexanders-MacBook-Pro-2 erb % npm run lint    

> lint
> cross-env NODE_ENV=development eslint . --cache --ext .js,.jsx,.ts,.tsx


/Users/alexanderolsen/workspace/test/erb/src/main/main.ts
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/extensions
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/no-duplicates
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/order
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/no-self-import
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/no-cycle
   1:1   error  Resolve error: TSError: ⨯ Unable to compile TypeScript:
.erb/configs/webpack.config.renderer.dev.ts:21:29 - error TS2533: Object is possibly 'null' or 'undefined'.

21 const requiredByDLLConfig = module.parent.filename.includes(
                               ~~~~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:100:5 - error TS2322: Type 'DllReferencePlugin | null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.
  Type 'null' is not assignable to type '((this: Compiler, compiler: Compiler) => void) | WebpackPluginInstance'.

100     requiredByDLLConfig
        ~~~~~~~~~~~~~~~~~~~
101       ? null
    ~~~~~~~~~~~~
... 
105           sourceType: 'var',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
106         }),
    ~~~~~~~~~~
.erb/configs/webpack.config.renderer.dev.ts:170:45 - error TS2345: Argument of type 'number | null' is not assignable to parameter of type 'number | undefined'.
  Type 'null' is not assignable to type 'number | undefined'.

170         .on('close', (code) => process.exit(code))
                                                ~~~~

    at createTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/alexanderolsen/workspace/test/erb/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)  import/no-useless-path-segments
  18:33  error  Missing file extension for "./utils"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              import/extensions

✖ 7 problems (7 errors, 0 warnings)

**Comment by amilajack** [comment]
I'm unable to repro. CI isn't failing either. Are you starting with a fresh clone of the project?

**Comment by aolsenjazz** [comment]
Yep I'm starting with a fresh clone. Note that the output I pasted here is the output of when an import is broken. See below for steps. Tested on both macOS 11.6 and Windows 11, virtually same output on both boxes.

````
git clone --depth 1 --branch main https://github.com/electron-react-boilerplate/electron-react-boilerplate.git erb
cd erb
npm i
````

break an import. I chose to break the util import in main.ts by changes file path to `./utils`

````
npm run lint
````

which should give you that excessive error output
